server/app/src/test/java/org/apache/james/app/spring/JamesSpringContextTest.java

```diff
@@ -0,0 +1,68 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.app.spring;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import org.apache.james.container.spring.context.JamesServerApplicationContext;
+import org.apache.james.mailbox.MailboxListener;
+import org.apache.james.mailbox.store.event.DefaultDelegatingMailboxListener;
+import org.apache.james.mailbox.store.quota.ListeningCurrentQuotaUpdater;
+
+import org.assertj.core.api.Condition;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+
+public class JamesSpringContextTest {
+    private static final Condition<MailboxListener> QUOTA_UPDATER_LISTENER = new Condition<MailboxListener>() {
+        @Override
+        public boolean matches(MailboxListener mailboxListener) {
+            return mailboxListener instanceof ListeningCurrentQuotaUpdater;
+        }
+    };
+    private static final int ONCE = 1;
+    private JamesServerApplicationContext context;
+
+    @Before
+    public void setup() throws Exception {
+        context = new JamesServerApplicationContext(new String[] { "META-INF/org/apache/james/spring-server.xml" });
+        context.registerShutdownHook();
+        context.start();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        context.stop();
+        context.destroy();
+    }
+
+    @Test
+    public void springShouldLoadAndAddOnlyOneQuotaUpdaterListener() {
+        Assume.assumeThat(System.getProperty("java.version"), CoreMatchers.startsWith("1.6"));
```

Comment:
I've just double checked, it works on Java 8 so I remove that assumption.
For you question, why did it has that assumption.
- At the first time, the testing for spring server load context was ignore because of [https://issues.apache.org/jira/browse/JAMES-1555](url). It did not consistence 
- Later on, we implemented annotation but it did not support fully on JPA. It made an error on java8 when loading spring context (Sorry, I did not remember well what was the exact issue), that 's why an assumption is added.